### PR TITLE
Feature/GitHub actions

### PR DESCRIPTION
**Pull request by ajax-yurkovskyi-t**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -55,6 +59,22 @@ kotlin {
     }
 }
 
+
 tasks.withType<Test> {
     useJUnitPlatform()
 }
+
+configure<DeltaCoverageConfiguration> {
+    val targetBranch = project.properties["diffBase"]?.toString() ?: "refs/remotes/origin/main"
```
Comment (ajax-torbin-v) [inline build.gradle.kts:67]: Why are you using main? 😡
Under conventions, you must use master

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
```
Comment (ajax-torbin-v) [inline .github/workflows/ci.yml:25]: You should use corretto

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```
Comment (ajax-torbin-v) [inline .github/workflows/ci.yml:24]: Why not 21 lts?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Kotlin CI with Gradle
```
Comment (ajax-torbin-v) [inline .github/workflows/ci.yml:1]: Naming is confusing

File: build.gradle.kts [hunk 5]
```diff
@@ -58,3 +62,18 @@ kotlin {
...
+    diffSource.byGit {
+        compareWith(targetBranch)
+    }
+
+    violationRules.failIfCoverageLessThan(0.6)
+    reports {
+        html.set(true)
+    }
+}
+tasks.check {
```
Comment (ajax-marchenko-s) [inline build.gradle.kts]: remove this dependsOn because you run deltacoverage twice

**Review by ajax-marchenko-s — COMMENTED** [review]
Great job

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ajax-yurkovskyi-t) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml (hunk 4 above)
Comment (ajax-yurkovskyi-t) [reply .github/workflows/ci.yml]: Fixed

File: build.gradle.kts (hunk 5 above)
Comment (ajax-yurkovskyi-t) [reply build.gradle.kts]: Done

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          java-version: '17'
+          distribution: 'corretto'
+          cache: gradle
+
+      - name: Run gradle build
+        run: ./gradlew build
+
+      - name: Run Delta Coverage check
+        if: ${{ github.event_name == 'pull_request' && ( success() || failure() ) }}
```
Comment (ajax-shulepa-a) [inline .github/workflows/ci.yml:32]: do you really need a coverage report when the build fails?

Comment (ajax-yurkovskyi-t) [reply .github/workflows/ci.yml:36]: this is what Sergii writes and IMO I think it can be useful

**Pull request merged**